src/sphinxcontrib/towncrier/__init__.py

```diff
@@ -105,6 +105,21 @@ def _get_draft_version_fallback(strategy: str, sphinx_config: Dict[str, Any]):
     return ' '.join(msg_chunks)
 
 
+def _nodes_from_rst(state: statemachine.State, rst: str) -> List[nodes.Node]:
```

Comment:
How do you feel about making this a directive method since it accesses `self.state` anyway? A disadvantage of this would be making the testing harder (dependency injection vs mocking).

Reply:
Yep, I saw the disadvantage of doing this, and not the advantage :)

All in all, testable functions are really interested. Plus, methods are great when you control the class, and can instantiate an object easily. Here, we don’t’ control SphinxDirectives, and Directives are really hard to instantiate, so I’d much rather follow the test pyramid and test as little as possible in integration and as much as possible in unit tests (well, unit-ish, as we’ll have to do some docutils in the tests). So it’s your code, it’s your choice, but I’d advocate largely for the function.

Reply:
Yeah, let's keep it separate then. Just wanted to hear more thoughts...